`eris keys ls` crashing when no keys present

before doing the first `eris keys export ...` the command `eris keys ls` leads to the following crash:

# eris keys ls
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/eris-ltd/eris-cli/keys.ListKeys(0xc8200f6e00, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/keys/readers.go:30 +0x1028
github.com/eris-ltd/eris-cli/cmd.ListKeys(0xf35020, 0xf793a0, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/keys.go:156 +0xe3
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).execute(0xf35020, 0xf793a0, 0x0, 0x0, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:569 +0x869
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).ExecuteC(0xf33020, 0xf35020, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:656 +0x56b
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).Execute(0xf33020, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:615 +0x2d
github.com/eris-ltd/eris-cli/cmd.Execute()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/eris.go:71 +0x32
main.main()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/eris/main.go:8 +0x14

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 18 [select]:
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.readLoop()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger_old.go:118 +0x1fb
created by github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.init.1
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger_old.go:30 +0x2b

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 21 [IO wait]:
net.runtime_pollWait(0x7f4c8676aa88, 0x72, 0xc82000e1a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201394f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201394f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820139490, 0xc820148000, 0x1000, 0x1000, 0x0, 0x7f4c885b6050, 0xc82000e1a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200a00c0, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7f4c8676ac08, 0xc8200a00c0, 0xc8200def78, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820134a60, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc820091a40)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc820091a40, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200def20)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0xc8200def20)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d
root@mobiatm:~# eris keys gen
A03A4130588A95184B58E016675577CE9A09FAB8
root@mobiatm:~# eris keys ls
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/eris-ltd/eris-cli/keys.ListKeys(0xc8200f6e00, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/keys/readers.go:30 +0x1028
github.com/eris-ltd/eris-cli/cmd.ListKeys(0xf35020, 0xf793a0, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/keys.go:156 +0xe3
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).execute(0xf35020, 0xf793a0, 0x0, 0x0, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:569 +0x869
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).ExecuteC(0xf33020, 0xf35020, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:656 +0x56b
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra.(*Command).Execute(0xf33020, 0x0, 0x0)
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/spf13/cobra/command.go:615 +0x2d
github.com/eris-ltd/eris-cli/cmd.Execute()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/eris.go:71 +0x32
main.main()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/cmd/eris/main.go:8 +0x14

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1696 +0x1

goroutine 19 [syscall]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 18 [select]:
github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.readLoop()
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger_old.go:118 +0x1fb
created by github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log.init.1
	/home/coda/sites/gostuff/src/github.com/eris-ltd/eris-cli/Godeps/_workspace/src/github.com/eris-ltd/common/go/log/logger_old.go:30 +0x2b

goroutine 21 [IO wait]:
net.runtime_pollWait(0x7efe16b4c9b8, 0x72, 0xc82000e1a0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201374f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201374f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820137490, 0xc820148000, 0x1000, 0x1000, 0x0, 0x7efe18998050, 0xc82000e1a0)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc8200a00c0, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
net/http.noteEOFReader.Read(0x7efe16b4cb38, 0xc8200a00c0, 0xc8200df918, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc820132a40, 0xc820148000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc82008f980)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc82008f980, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc8200df8c0)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0xc8200df8c0)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

--- a/keys/readers.go
+++ b/keys/readers.go
@@ -22,15 +22,19 @@ func ListKeys(do *definitions.Do) error {
 		if err != nil {
 			return err
 		}
-		hostAddrs := make([]string, len(addrs))
-		for i, addr := range addrs {
-			hostAddrs[i] = addr.Name()
-		}
-		do.Result = strings.Join(hostAddrs, ",")
-		log.WithField("=>", hostAddrs[0]).Warn("The keys on your host kind marmot:")
-		hostAddrs = append(hostAddrs[:0], hostAddrs[1:]...)
-		for _, addr := range hostAddrs {
-			log.WithField("=>", addr).Warn()
+		if len(addrs) == 0 {
+			log.Warn("No keys found on host.")
+		} else {
+			hostAddrs := make([]string, len(addrs))
+			for i, addr := range addrs {
+				hostAddrs[i] = addr.Name()
+			}
+			do.Result = strings.Join(hostAddrs, ",")
+			log.WithField("=>", hostAddrs[0]).Warn("The keys on your host kind marmot:")
+			hostAddrs = append(hostAddrs[:0], hostAddrs[1:]...)
+			for _, addr := range hostAddrs {
+				log.WithField("=>", addr).Warn()
+			}
 		}
 	}
 
@@ -51,12 +55,15 @@ func ListKeys(do *definitions.Do) error {
 		}
 		keysOutString := strings.Split(util.TrimString(string(keysOut.Bytes())), "\n")
 		do.Result = strings.Join(keysOutString, ",")
-		log.WithField("=>", keysOutString[0]).Warn("The keys in your container kind marmot:")
-		keysOutString = append(keysOutString[:0], keysOutString[1:]...)
-		for _, addr := range keysOutString {
-			log.WithField("=>", addr).Warn()
+		if len(keysOutString) == 0 || keysOutString[0] == "" {
+			log.Warn("No keys found in container.")
+		} else {
+			log.WithField("=>", keysOutString[0]).Warn("The keys in your container kind marmot:")
+			keysOutString = append(keysOutString[:0], keysOutString[1:]...)
+			for _, addr := range keysOutString {
+				log.WithField("=>", addr).Warn()
+			}
 		}
-
 	}
 	return nil
 }

